File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        crr=head
        s=""
        while crr!=None:
            s=s+str(crr.val)
            crr=crr.next
        return int(s,2)

File: 1411-convert-binary-number-in-a-linked-list-to-integer/convert-binary-number-in-a-linked-list-to-integer.c
int getDecimalValue(struct ListNode* head) {

        int len=0;
        struct ListNode* crr=head;

        while(crr!=NULL){
            len++;
            crr=crr->next;
        }

        int p=len-1;
        int sum=0;
        crr=head;
        while(crr!=NULL){
            sum=sum+(crr->val*pow(2,p));
            crr=crr->next;
            p--;
        }
        return sum;
}